File: packages/ios/Sources/Components/Drawer/Drawer.figma.swift
import SwiftUI
import Figma

struct KozmosDrawerConnect: FigmaConnect {
    let component = KozmosDrawer<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=232-2042"

    @FigmaEnum(
        "Side",
        mapping: [
            "Top": KozmosDrawerSide.top,
            "Right": KozmosDrawerSide.right,
            "Bottom": KozmosDrawerSide.bottom,
            "Left": KozmosDrawerSide.left
        ]
    )
    var side: KozmosDrawerSide = .right

    @FigmaString("Title Text")
    var title: String = "Manage layers"

    @FigmaString("Description Text")
    var description: String = "Update map display settings."

    @FigmaString("Body Text")
    var bodyText: String = "Drawer body content"

    var body: some View {
        KozmosDrawer(
            isPresented: .constant(true),
            side: self.side,
            title: self.title,
            description: self.description,
            bodyText: self.bodyText
        ) {
            AnyView(Spacer(minLength: 0))
        }
    }
}

File: packages/android/src/main/java/com/kozmos/components/Drawer/Drawer.figma.kt
package com.kozmos.components.drawer

import androidx.compose.foundation.layout.Spacer
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=232-2042")
class KozmosDrawerConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Manage layers"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Update map display settings."

    @FigmaProperty(FigmaType.Text, "Body Text")
    val bodyText: String = "Drawer body content"

    @FigmaProperty(FigmaType.Enum, "Side")
    val side: KozmosDrawerSide = Figma.mapping(
        "Top" to KozmosDrawerSide.Top,
        "Right" to KozmosDrawerSide.Right,
        "Bottom" to KozmosDrawerSide.Bottom,
        "Left" to KozmosDrawerSide.Left
    )

    @Composable
    fun ComponentExample() {
        KozmosDrawer(
            open = true,
            onDismissRequest = {},
            side = side,
            drawerContent = {
                DrawerHeader(
                    title = title,
                    description = description,
                    showCloseButton = false,
                    onDismissRequest = {}
                )
                Text(bodyText)
                Spacer(modifier = Modifier.weight(1f))
            }
        ) {
            Text("Screen content")
        }
    }
}
